tools/src/main/scala/com/nvidia/spark/rapids/tool/planparser/HashAggregateExecParser.scala

```diff
@@ -37,10 +37,9 @@ case class HashAggregateExecParser(
     val maxDuration = SQLPlanParser.getTotalDuration(accumId, app)
     val exprString = node.desc.replaceFirst("HashAggregate", "")
     val expressions = SQLPlanParser.parseAggregateExpressions(exprString)
-    val isAllExprsSupported = expressions.forall(expr => checker.isExprSupported(expr))
     val notSupportedExprs = expressions.filter(expr => !checker.isExprSupported(expr))
     val (speedupFactor, isSupported) = if (checker.isExecSupported(fullExecName) &&
-        isAllExprsSupported) {
+        notSupportedExprs.size == 0) {
```

Comment:
here and elsewhere isEmpty is idiomatic and  more efficient https://stackoverflow.com/questions/31979945/scala-intellij-suggests-replacing-size-check-with-non-empty-but-why
```suggestion
        notSupportedExprs.isEmpty) {
```

also the code appears to be quite repetitive and could use some refactoring.